Store collaborative local git metrics

salmavkh opened the pull request:
## 📝 Description

This PR stores metrics obtained from local .git file detection for code collaborative:
- Created code_collaborative_metrics and code_collaborative_summary tables to store collaborative code git metrics and both manual/LLM summaries.
- Added insert_code_collaborative_metrics, get_metrics_id, and insert_code_collaborative_summary helpers and wire them into analyze_code_project so collaborative code runs persist local git metrics and summaries.
- Added tests to cover inserting and updating metrics, handling missing/partial sections safely, and storing both non-LLM and LLM summaries for the same project.

**Closes:** #254 

---

## 🔧 Type of Change

- [ ] 🐛 Bug fix (non-breaking change that fixes an issue)
- [x] ✨ New feature (non-breaking change that adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] 📚 Documentation added/updated
- [x] ✅ Test added/updated
- [ ] ♻️ Refactoring
- [ ] ⚡ Performance improvement

---

## 🧪 Testing

- [x] pytest
- [x] test with code collaborative (llm and non-llm), ensure it stores all metrics correctly in the db

---

## ✓ Checklist

- [x] 🤖 GenAI was used in generating the code and I have performed a self-review of my own code
- [x] 💬 I have commented my code where needed
- [x] 📖 I have made corresponding changes to the documentation
- [x] ⚠️ My changes generate no new warnings
- [x] ✅ I have added tests that prove my fix is effective or that my feature works and tests are passing locally
- [x] 🔗 Any dependent changes have been merged and published in downstream modules
- [ ] 📱 Any UI changes have been checked to work on desktop, tablet, and/or mobile

---

## 📸 Screenshots

<details>
<summary>Click to expand screenshots</summary>

<img width="1732" height="1048" alt="image" src="https://github.com/user-attachments/assets/196faf24-d2be-4391-8997-da401efcc68a" />

<img width="1704" height="1034" alt="image" src="https://github.com/user-attachments/assets/b5883d56-e83e-4cf9-b406-0bd1fa6a6d47" />

</details>

salmavkh committed:
added local_git_metrics_collaborative table

salmavkh committed:
added store_local_git_metrics_collaborative()

salmavkh committed:
put the function in analyze_code_project()

salmavkh committed:
added a test file

salmavkh committed:
moved storing after user input desc

ammaarkhan reviewed (changes requested):
Hey Salma, great work on this. The code changes are clean and easy to follow. 

Some minor issues I noticed:
1. The tests are currently following happy path cases. There is no coverage for missing / partial metrics so empty values can slip through. Could we add some additional test cases regarding this?
2. Upon testing locally I noticed the frameworks detection and desc fields are currently empty. These are likely unrelated bugs (Timmi flagged the frameworks detection not working last week), but wanted to bring it up just in case.
<img width="469" height="84" alt="Image" src="https://github.com/user-attachments/assets/35abf234-f9e2-47a4-80c0-9448f3dbbff7" />

salmavkh committed:
updated the db structure

salmavkh committed:
integrated into the project flow

salmavkh committed:
updated tests

salmavkh commented:
@ammaarkhan Thanks for the review! I’m also not sure why the framework detection isn’t working.

For the description field, it was originally just meant for non-LLM user input, but then I realized we also need to store the LLM summary. So I ended up splitting it into two tables: for code collaboration metrics (.git file) and for summaries (both user-input and LLM). I also added some non-happy-path tests based on your suggestions.

I updated the PR description too. Sorry about that, and let me know if it looks good!

taoTimTim reviewed (changes requested):
Good work.

All tests are passing.

The code is good, everything is being stored properly:
<img width="1882" height="622" alt="image" src="https://github.com/user-attachments/assets/47a67d18-590e-4b6b-b371-7da8d5b2e88e" />

I like how you separated the metrics into two tables, the metrics themselves and the summary.

I made one inline comment below, just to keep the db queries separate from any business logic.

salmavkh committed:
moved logic from src/db and updated tests

salmavkh commented:
@taoTimTim oops sorry about that, won’t happen again, i should’ve been more careful. thanks for the review! feel free to take another look

taoTimTim reviewed (approved):
No worries, Salma! I've caught myself doing it too. Thanks for changing that :) Approved.

ammaarkhan reviewed (approved):
Thanks for making the changes Salma! The test cases are better and two database structure looks good. All tests pass and the summary table is storing the info accurately. Great work. 

<img width="850" height="127" alt="Image" src="https://github.com/user-attachments/assets/baf77ee0-8284-4f61-a999-e647d21dbf38" />

<img width="1037" height="495" alt="Image" src="https://github.com/user-attachments/assets/b0dcf42f-27ce-4fe5-99c6-6eb6ce201cf9" />

salmavkh committed:
Merge branch 'main' into store-collaborative-local-git-metrics
